admiral/pkg/clusters/DRUtil.go

```diff
@@ -0,0 +1,47 @@
+package clusters
+
+import (
+	"context"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/controller/common"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+const  READ_WRITE_ENABLED = false
+const READ_ONLY_ENABLED = true;
+
+type AdmiralState struct {
+	ReadOnly  bool
+}
+
+type AdmiralStateChecker interface {
+	runStateCheck(as *AdmiralState,ctx context.Context)
+	shouldRunOnIndependentGoRoutine() bool
+}
+/*
+Utility function to start Admiral DR checks.
+DR checks can be run either on the main go routine or a new go routine
+*/
+func RunAdmiralStateCheck(asc AdmiralStateChecker, as *AdmiralState,ctx context.Context){
```

Comment:
nitpick, if possible, can we make the `ctx` the first param?

Reply:
This is fixed